core/trace_point.rbs

```diff
@@ -172,7 +172,7 @@ class TracePoint < Object
   # (note that we needed to filter out calls by itself from :line handler,
   # otherwise it will call itself infinitely).
   #
-  def self.allow_reentry: () { () -> void } -> void
+  def self.allow_reentry: [T] () { (nil) -> T } -> T
```

Comment:
While the documentation says the block takes nothing, it actually is given a single `nil` as an argument